pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer_test.go

```diff
@@ -471,6 +471,152 @@ func TestApplyLocalitySetting(t *testing.T) {
 					},
 				},
 			},
+			{
+				name:             "priority assignment as per the overridden value",
+				failoverPriority: []string{"topology.istio.io/network=n1"},
+				proxyLabels: map[string]string{
+					"topology.istio.io/network": "n2",
+					"topology.istio.io/cluster": "test",
+				},
+				expected: []*endpoint.LocalityLbEndpoints{
+					{
+						Locality: &core.Locality{
+							Region:  "region1",
+							Zone:    "zone1",
+							SubZone: "subzone1",
+						},
+						LbEndpoints: []*endpoint.LbEndpoint{
+							{
+								HostIdentifier: buildEndpoint("1.1.1.1"),
+								LoadBalancingWeight: &wrappers.UInt32Value{
+									Value: 1,
+								},
+							},
+						},
+						LoadBalancingWeight: &wrappers.UInt32Value{
+							Value: 1,
+						},
+						Priority: 0,
+					},
+					{
+						Locality: &core.Locality{
+							Region:  "region1",
+							Zone:    "zone1",
+							SubZone: "subzone1",
+						},
+						LbEndpoints: []*endpoint.LbEndpoint{
+							{
+								HostIdentifier: buildEndpoint("2.2.2.2"),
+								LoadBalancingWeight: &wrappers.UInt32Value{
+									Value: 1,
+								},
+							},
+						},
+						LoadBalancingWeight: &wrappers.UInt32Value{
+							Value: 1,
+						},
+						Priority: 1,
+					},
+					{
+						Locality: &core.Locality{
+							Region:  "region2",
+							Zone:    "zone2",
+							SubZone: "subzone2",
+						},
+						LbEndpoints: []*endpoint.LbEndpoint{
+							{
+								HostIdentifier: buildEndpoint("3.3.3.3"),
+								LoadBalancingWeight: &wrappers.UInt32Value{
+									Value: 1,
+								},
+							},
+						},
+						LoadBalancingWeight: &wrappers.UInt32Value{
+							Value: 1,
+						},
+						Priority: 0,
+					},
+					{
+						Locality: &core.Locality{
+							Region:  "region2",
+							Zone:    "zone2",
+							SubZone: "subzone2",
+						},
+						LbEndpoints: []*endpoint.LbEndpoint{
+							{
+								HostIdentifier: buildEndpoint("4.4.4.4"),
+								LoadBalancingWeight: &wrappers.UInt32Value{
+									Value: 1,
+								},
+							},
+						},
+						LoadBalancingWeight: &wrappers.UInt32Value{
+							Value: 1,
+						},
+						Priority: 1,
+					},
+				},
+			},
+			{
+				name:             "no endpoints with overridden value",
+				failoverPriority: []string{"topology.istio.io/network=n3"},
```

Comment:
can we have a test like `{"topology.istio.io/network", "topology.istio.io/network=n3"}`? I think that should mean "prefer same network, then network=n3", right?

Reply:
I think `{"topology.istio.io/network", "topology.istio.io/network=n3"}` will be effectively `{"topology.istio.io/network"}`. 

As per the current logic,
> Endpoints matching all N labels with the client proxy have priority P(0) i.e. the highest priority.
> Endpoints matching the first N-1 labels with the client proxy have priority P(1) i.e. second highest priority.

For the endpoints matching of N th label to be considered, all the preceeding labels i.e., N-1 labels should also match. Please cmiiw.

Reply:
ah good point